app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -3025,15 +3025,17 @@ class BraveVPNService :
                 ProxyManager.ID_WG_BASE + id
             } else if (WireguardManager.catchAllEnabled()) {
                 // if the enabled wireguard is catchall-wireguard, then return wireguard id
-                val id = WireguardManager.getOptimalCatchAllConfigId() ?: return Backend.Base
+                val endpoint = appConfig.getSelectedDnsProxyDetails()
```

Comment:
The endpoint address should be fetched from the Go library to obtain the exact IP. I will merge this and make those changes.

Reply:
Just testing this some more and I see another issue. It's likely resolved by your comment about using the Go library to get the exact IP, but I'm not sure so I'll mention it here. When I set DNS to use DOH, `appConfig.getSelectedDnsProxyDetails()` returns null which is then passed into my modified `getOptimalCatchAllConfigId` which skips the canRoute check which causes the original issue again. So we should probably require a non-null IP be passed to this function (since DNS IP can't be null anyway). And hopefully your way of getting the DNS IP won't ever return null anyway. Cheers